src/ui/theme/utils.ts

```diff
@@ -0,0 +1,224 @@
+  };
+};
+
+const textColorsForBackground = (
+  backgroundColor: string,
+  primaryColor: string,
+  defaultColors: Colors,
+) => {
+  const textColors = {
+    "grey-text-dark": defaultColors["grey-text-dark"],
+    "grey-text-light": defaultColors["grey-text-light"],
+    "grey-ui-dark": defaultColors["grey-ui-dark"],
+    "grey-ui-light": defaultColors["grey-ui-light"],
+    "primary-text": defaultColors["primary-text"],
+  };
+
+  // Find the text colors for the background
+  if (backgroundColor?.startsWith("#")) {
+    const rgb = hexToRGB(backgroundColor);
+    if (rgb !== null) {
+      Object.assign(textColors, rgbToTextColors(rgb));
+    }
+  }
+
+  // Find the text color for the primary color
+  if (primaryColor?.startsWith("#")) {
+    const rgb = hexToRGB(primaryColor);
+    if (rgb !== null) {
+      textColors["primary-text"] = isLightColor(rgb) ? "black" : "white";
+    }
+  }
+
+  return textColors;
+};
+
+const fallback = (somethingNullable: any | null, defaultValue: any) => {
+  return somethingNullable ? somethingNullable : defaultValue;
+};
+
+const mapColors = (
+  colorsMapping: any,
+  defaultColors: Colors,
+  brandingAppearance?: BrandingAppearance | undefined,
+): Colors => {
+  const mappedColors = Object.entries(colorsMapping).map(([target, source]) => [
+    target,
+    fallback(
+      brandingAppearance
+        ? brandingAppearance[source as keyof BrandingAppearance]
+        : null,
+      defaultColors[target as keyof Colors],
+    ),
+  ]);
+
+  return Object.fromEntries(mappedColors) as Colors;
+};
+
+export const toColors = (
+  colorsMapping: any,
+  defaultColors: Colors,
+  brandingAppearance?: BrandingAppearance | undefined,
+): Colors => {
+  const mappedColors = mapColors(
+    colorsMapping,
+    defaultColors,
+    brandingAppearance,
+  );
+
+  return brandingAppearance
+    ? {
+        ...defaultColors,
+        ...mappedColors,
+        ...textColorsForBackground(
+          mappedColors.background,
+          mappedColors.primary,
+          defaultColors,
+        ),
+      }
+    : { ...defaultColors }; //copy, do not reference.
+};
+
+export const toProductInfoColors = (
+  brandingAppearance?: BrandingAppearance | undefined,
+): Colors => {
+  return toColors(
+    InfoColorsToBrandingAppearanceMapping,
+    DEFAULT_INFO_COLORS,
+    brandingAppearance,
+  );
+};
+
+export const toFormColors = (
+  brandingAppearance?: BrandingAppearance | undefined,
+): Colors => {
+  return toColors(
+    FormColorsToBrandingAppearanceMapping,
+    DEFAULT_FORM_COLORS,
+    brandingAppearance,
+  );
+};
+
+export const toShape = (
+  brandingAppearance?: BrandingAppearance | undefined,
+): Shape => {
+  if (!brandingAppearance) {
+    return DefaultShape;
+  }
+  switch (brandingAppearance.shapes) {
+    case "rounded":
+      return RoundedShape;
+    case "rectangle":
+      return RectangularShape;
+    case "pill":
+      return PillsShape;
+    default:
```

Comment:
FFTI. Hmm I'm wondering if we could change the type of shapes to an `enum` so we can have exhaustive switch-cases... NABD right now though.

Reply:
Yup makes sense 👍